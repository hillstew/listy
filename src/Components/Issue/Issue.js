import React from 'react';

const Issue = (props) => {
  const { id, body } = props.issue;
  return (
    <li key={id} id={id} className="issue-li">
      <span>
        <i onClick={props.toggleIssueCompletion} className="fas fa-square" />
        <input
          onChange={props.handleBodyChange}
          placeholder='Add a list item...'
          value={body}
        />
      </span>
      <button onClick={props.removeIssue} className="delete-button"/>
    </li>
  )
}

export default Issue;