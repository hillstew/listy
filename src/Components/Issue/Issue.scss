.issue-li {
  border-bottom: 1px solid grey;
  margin: 8px 0;

  i {
    cursor: pointer;
    padding: 5px;
  }

  input {
    border: transparent;
  }
}

.list-delete-button {
  background: url(./images/delete.png);
  background-size: cover;
  border: none;
  margin-top: 7px;
  height: 14px;
  width: 14px;

  &:hover {
    background: url(./images/delete-green.png);
    background-size: cover;
  }
}

.incomplete-item {
  background: url(./images/square.png);
  background-size: cover;
  margin-top: 5px;
  margin-right: 5px;
  border: none;
  height: 16px;
  width: 16px;

  &:hover {
    background: url(./images/square-green.png);
    background-size: cover;
  }
}

.complete-item {
  background: url(./images/checkbox.png);
  background-size: cover;
  margin-top: 5px;
  margin-right: 5px;
  border: none;
  height: 16px;
  width: 16px;

  &:hover {
    background: url(./images/checkbox-green.png);
    background-size: cover;
  }
}