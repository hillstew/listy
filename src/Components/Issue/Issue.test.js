import React from 'react'
import Issue from './Issue';
import { shallow } from 'enzyme'

describe('Issue', () => {
    let wrapper;
    const mockToggleIssueCompletion = jest.fn()
    const mockHandleBodyChange = jest.fn()
    const mockRemoveIssue = jest.fn()

    beforeEach(() => {
        let props = {
            issue: {
                id: 1,
                body: '',
            }
        }
        wrapper = shallow(<Issue {...props} toggleIssueCompletion={mockToggleIssueCompletion} handleBodyChange={mockHandleBodyChange} removeIssue={mockRemoveIssue} />)
    });

    it('should match the correct snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
});
