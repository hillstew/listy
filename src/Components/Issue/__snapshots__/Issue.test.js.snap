// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue should match the correct snapshot 1`] = `
<li
  className="issue-li"
  id={1}
  key="1"
>
  <span>
    <button
      className="incomplete-item"
      onClick={[MockFunction]}
    />
    <input
      maxlength="45"
      onChange={[MockFunction]}
      placeholder="Add a list item..."
      value=""
    />
  </span>
  <button
    className="list-delete-button"
    onClick={[MockFunction]}
  />
</li>
`;
